### ISSUE TITLE
Different GPU memory usage with different GPUs

### ISSUE THREAD

**Issue by FlySka** [ultralytics/ultralytics#6717]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Predict

### Bug

Hello,

I don't know if this is a bug or not. I noticed with my team that the YoloV8 model has different GPU memory usage depending on the GPU model used. We are not sure if it is because of the GPU itself or because of the CUDA version or another reason.

The detail is this:

1. Environment 1:
    - CPU: AMD Ryzen 7 4800H
    - GPU: NVIDIA GeForce GTX 1650 Ti mobile
    - RAM: 16GB
    - SO: Ubuntu 20.04
    - Python: 3.8.10
    - Torch: 2.0.1
    - Torch-vision: 0.15.2
    - CUDA: 11.7
    - cuDNN: 8.8.1

     -----------> **Memory GPU Usage 1377MiB**

2. Environment 2:
    - CPU: Intel i9-13900HX
    - GPU: NVIDIA GeForce RTX 4060
    - RAM: 16GB
    - SO: Ubuntu 22.04
    - Python: 3.10.12
    - Torch: 2.1.0
    - Torch-vision: 0.16.0
    - CUDA: 12.1
    - cuDNN: 8.9.2

     -----------> **Memory GPU Usage 450MiB**

Regarding this, I don't know if you could give us a solution in case it is a bug or if you could give us a reason for this difference in memory usage.

pd: the YoloV8s model of object detection was used.

### Minimal Reproducible Example

Code for extracting cuda version from torch

```
import torch
print(torch.backends.cudnn.version())
print(torch.version.cuda)
```

Simple code to test memory usage:

```
import os
import cv2
import ultralytics

NEURON_PATH = 'data/anomalias_yolov8s_25-10-23.pt'
IMAGE_PATH = "data/test.jpg"
batch_test = 1

detector = ultralytics.YOLO(NEURON_PATH)

# Creamos un batch de test
images_path = []
for n in range(bastch_test):
    images_path.append(IMAGE_PATH)

# Cargamos la imagen
images = []
images_resize = []
for i in images_path:
    image = cv2.imread(i)
    image_resize = cv2.resize(deepcopy(image), (640, 640))
    images.append(image)
    images_resize.append(image_resize)

# Hacemos la prediccion
print("Inferencia...")
 if bastch_test > 0:
    while True:
        predictions = detector.predict(images_resized, imgsz=640, conf=0.2 iou=0.4, verbose=False)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@FlySka hello,

The differences in GPU memory usage you're observing across different environments are likely caused by several factors rather than a bug with YOLOv8. Here are a few points to consider:

1. **GPU Architecture**: Different GPUs have varying architectures, which affect how memory is managed and utilized. This also includes the efficiency of memory usage for certain operations.

2. **CUDA/CuDNN Version**: Different versions of CUDA and cuDNN can have optimizations that reduce memory usage. Newer versions typically include improvements and enhancements that can lead to lower memory consumption.

3. **PyTorch & Torchvision Versions**: These libraries are updated regularly, with each version potentially bringing performance improvements and changes in memory management.

4. **Software Stack**: The combination of Operating System, Python, and the CUDA toolkit can also influence memory usage.

Since the GPUs in question are from different families (GTX vs. RTX), it's conceivable that the RTX 4060 with newer architecture and updated software stack is more efficient with memory usage during inference. Each GPU may allocate memory differently based on its internal optimizations and the drivers it uses.

In conclusion, the observed behavior is expected and within the norm of running inference on different hardware and software configurations. 🤖

If you require further assistance, feel free to check our documentation or continue the discussion here. Thank you for contributing to the YOLOv8 community! 🚀

**Comment by glenn-jocher** [comment]
Hello @FlySka,

While we don't have published studies specific to YOLOv8 regarding memory performance across different GPU families and software versions, it's generally observed that newer software versions tend to bring performance improvements, including more efficient memory usage. However, the actual impact varies based on the specific combination of GPU, drivers, and software stack.

If it's not feasible to perform tests in your production environment, you might consider benchmarks from the broader AI and ML community or from the manufacturers (NVIDIA) that often include memory and performance analyses for different configurations.

Keeping your software stack updated, especially CUDA and cuDNN, is generally recommended to take advantage of the latest optimizations and enhancements for your hardware.

🌟 To stay attuned with the most recent optimization trends, I'd advise your team to regularly monitor the updates and release notes from software and hardware providers.📈

Warm
